strawberryfields/engine.py

```diff
@@ -501,7 +509,8 @@ def run(self, program, *, args=None, compile_options=None, **kwargs):
             # session and feed_dict are needed by TF backend both during simulation (if program
             # contains measurements) and state object construction.
             result._state = self.backend.state(**temp_run_options)
-
+            if self.backend_name == "bosonic":
+                result._ancilla_samples = self.backend.ancillae_samples_dict
```

Comment:
Just a heads up that `result._ancilla_samples` and `self.backend.ancillae_samples_dict` will be referencing the same dict (so changing one will change the other). This might not be an issue here, but perhaps safer to take a copy:

`result._ancilla_samples = self.backend.ancillae_samples_dict.copy()`

Reply:
Good catch!
```suggestion
                result._ancilla_samples = self.backend.ancillae_samples_dict.copy()
```